ee-feature-pack/galleon-shared/src/main/resources/feature_groups/remote-naming.xml

```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <feature-group-spec name="remote-naming" xmlns="urn:jboss:galleon:feature-group:1.0">
+    <feature-group name="remoting"/>
```

Comment:
I'm not sure about this aspect. Feature groups are just config chunks; layers are the things that have dependency relationships. This is a kind of a dependency relationship, and by duplicating the config chunk it runs the risk of falling out of sync with what the layers specify.

Reply:
There is no "remote-naming" layer - there is only a "remote-naming" feature group, which is not usable without remoting.  Do you suggest we create one?  Or do you mean to add the "remoting" feature group to every layer that uses the "remote-naming" feature group?

Reply:
The latter. IOW, instead of the deletion at https://github.com/wildfly/wildfly/pull/17145/files#diff-d27c5509481591a1bc0a83645c0fc7f9e507ad6dfb360b3d7ddd6138bed5eff4L4-L11, replace it with a simplified reference to the remoting feature-group.

The two layers that use remote-naming FG already have layer dependencies on the remoting layer.

The remoting feature-group provides config elements that if used mean other things are required, but it doesn't include those in its own definition. So this is no different.

Probably the embedded-activemq feature-group shouldn't embed remote naming either; it should be included  in the embedded-activemq layer. AFAICT the only uses of that feature-group are in the layer.

I realize this is kind of a gray area and TBH I'm thinking it through as I write. But perhaps it's like this...

The bulk of feature-groups are just config chunks. They provide a specific set of configuration but don't attempt to include other reusable things that their own chunk needs.

Then we have layers, which depend on other layers for things they need, and add feature groups / features for other things they need not provided by their dependencies.

Finally we have a subset of feature groups that are basically used to support construction of the OOTB configs, particularly for domain.xml. These are kind of a special case where we allow composition in a feature group, which, unlike standalone.xml we can't generate from layers. But these are more of a specialized use case.

@jfdenise @ehsavoie FYI re this discussion.

Reply:
I agree with the fact that the feature groups should be used in a layer as opposed to used by another feature group as from a provisioning perspective le granularity level should be the layer

Reply:
+1 @bstansberry , fg including other fg due to the requirement of layers doesn't help readability. Better to explicitly add multiple fg directly in the layers.

Reply:
> The latter. IOW, instead of the deletion at https://github.com/wildfly/wildfly/pull/17145/files#diff-d27c5509481591a1bc0a83645c0fc7f9e507ad6dfb360b3d7ddd6138bed5eff4L4-L11, replace it with a simplified reference to the remoting feature-group.
> 
> The two layers that use remote-naming FG already have layer dependencies on the remoting layer.
> 
> The remoting feature-group provides config elements that if used mean other things are required, but it doesn't include those in its own definition. So this is no different.
> 
> Probably the embedded-activemq feature-group shouldn't embed remote naming either; it should be included in the embedded-activemq layer. AFAICT the only uses of that feature-group are in the layer.

Understood now.  Will fix.

Reply:
Fixed.

Reply:
Thanks, @pferraro. I'll have a look.

FYI, Paul and @ehsavoie and @jfdenise this discussion sparked a thought so I filed https://issues.redhat.com/browse/WFLY-18440.